src/print/mod.rs

```diff
@@ -2111,6 +2111,99 @@ impl Print for Attr {
+            // FIXME(eddyb) move from repeating the same backtrace-like comments
+            // (potentially many times in a row) to a more "stateful" approach,
+            // which eonly mentions every inlined callsite once per sequence of
+            // e.g. `DataInst`s that are all nested in it.
+            &Attr::DbgSrcLoc(OrdAssertEq(dbg_src_loc)) => {
+                let mut comments = SmallVec::<[_; 8]>::new();
+                let mut next_dbg_src_loc = Some(dbg_src_loc);
+                while let Some(dbg_src_loc) = next_dbg_src_loc {
+                    let DbgSrcLoc {
+                        file_path,
+                        start_line_col: (start_line, mut start_col),
+                        end_line_col: (end_line, mut end_col),
+                        inlined_callee_name_and_call_site,
+                    } = dbg_src_loc;
+
+                    // HACK(eddyb) Rust-GPU's column numbers seem
+                    // off-by-one wrt what e.g. VSCode expects
+                    // for `:line:col` syntax, but it's hard to
+                    // tell from the spec and `glslang` doesn't
+                    // even emit column numbers at all!
+                    start_col += 1;
+                    end_col += 1;
+
+                    let mut s = String::new();
+                    if comments.is_empty() {
+                        s += "// at ";
+                    } else {
+                        s += "// … at ";
+                    }
+
+                    // HACK(eddyb) only skip string-quoting and escaping for
+                    // well-behaved file paths.
+                    let file_path = &printer.cx[file_path];
+                    if file_path.chars().all(|c| c.is_ascii_graphic() && c != ':') {
+                        s += file_path;
+                    } else {
+                        write!(s, "{file_path:?}").unwrap();
+                    }
+
+                    // HACK(eddyb) the syntaxes used are taken from VSCode, i.e.:
+                    // https://github.com/microsoft/vscode/blob/6b924c5/src/vs/workbench/contrib/terminalContrib/links/browser/terminalLinkParsing.ts#L75-L91
+                    // (using the most boring syntax possible for every situation).
+                    let is_quoted = s.ends_with('"');
+                    let is_range = (start_line, start_col) != (end_line, end_col);
+                    write!(
+                        s,
+                        "{}{start_line}{}{start_col}",
+                        if is_quoted { ',' } else { ':' },
+                        if is_quoted && is_range { '.' } else { ':' }
+                    )
+                    .unwrap();
+                    if is_range {
+                        s += "-";
+                        if start_line != end_line {
+                            write!(s, "{end_line}.").unwrap();
+                        }
+                        write!(s, "{end_col}").unwrap();
+                    }
```

Comment:
This is the more "subjective" part of this change (based on [comments in VSCode's source code](https://github.com/microsoft/vscode/blob/6b924c5/src/vs/workbench/contrib/terminalContrib/links/browser/terminalLinkParsing.ts#L75-L91), also linked in the `// HACK...` comment for posterity).

Since multiple syntaxes are possible for (almost) every situation, I tried to mix and match them to reduce variation or redundancy wherever possible.